books/tests/test_views.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+from django.test import RequestFactory
+from django.contrib.auth.models import AnonymousUser
+from mixer.backend.django import mixer
+
+pytestmark = pytest.mark.django_db
+
+from .. import views
+
+class TestHomePageView:
+    def test_anonymous(self):
+        req = RequestFactory().get('/')
```

Comment:
Instead of the `RequestFactory` (which is usually needed for more complicated cases) you can also use `client`. [Both are available as pytest fixtures too](
https://pytest-django.readthedocs.io/en/latest/helpers.html#rf-requestfactory).

`client` should allow you to log in as any user. [There are also fixtures that you can use if you want to test things as a super-user](https://pytest-django.readthedocs.io/en/latest/helpers.html#admin-client-django-test-client-logged-in-as-admin).

Reply:
Thanks for all this, that is really illuminating. I have just updated the BookNewVIew with tests both using CustomUser and the client and admin_client fixtures. The fixtures are fantastic, so they are just there, you don't even need to import anything it seems